### ISSUE TITLE
New cop: `Lint/UselessRescue`

### ISSUE THREAD

**Issue by fatkodima** [rubocop/rubocop#11359]
It should detect `rescue` clauses which are only reraising the catched error. This is possibly a bug, or just a useless code.

```ruby
# bad
def foo
  do_something
rescue SomeError
  raise
end

# bad
def foo
  do_something
rescue e
  raise e
end

# good
def foo
  do_something
rescue e
  do_something_more
  raise e
end
```

**Comment by vlad-pisanov** [comment]
nice! just be cautious of edge cases like:
```ruby
def foo
  do_something
rescue ArgumentError  # this rescue cannot be removed 🙅‍♂️
  raise
rescue
  puts 'oops'
end
```

**Comment by rporrasluc** [comment]
Hello @bbatsov I have found an edge case that I think it's reporting wrongly:

```ruby
def foo
  do_something
rescue => e
  raise
ensure
  puts e
end
```

`e` is used in the ensure block but the cop reports it as unused.

**Comment by koic** [comment]
@rporrasluc That issue will be resolved by https://github.com/rubocop/rubocop/pull/11423.

**Comment by rporrasluc** [comment]
> @rporrasluc That issue will be resolved by #11423.

Thank you @koic, my fault, I should have checked opened issues/PRs.
